src/ccmain/tessedit.cpp

```diff
@@ -81,17 +83,13 @@ bool Tesseract::init_tesseract_lang_data(const std::string &arg0,
                                          bool set_only_non_debug_params, TessdataManager *mgr) {
   // Set the language data path prefix
   lang = !language.empty() ? language : "eng";
-  language_data_path_prefix = datadir;
-  language_data_path_prefix += lang;
-  language_data_path_prefix += ".";
+  std::filesystem::path tessdata_path = datadir / (lang + "." + kTrainedDataSuffix);
 
   // Initialize TessdataManager.
-  std::string tessdata_path = language_data_path_prefix + kTrainedDataSuffix;
-  if (!mgr->is_loaded() && !mgr->Init(tessdata_path.c_str())) {
-    tprintf("Error opening data file %s\n", tessdata_path.c_str());
-    tprintf(
+  if (!mgr->is_loaded() && !mgr->Init(tessdata_path.string().c_str())) {
```

Comment:
```suggestion
  if (!mgr->is_loaded() && !mgr->Init(tessdata_path.c_str())) {
```

Reply:
This change would broke msvc build:
`error C2664: 'bool tesseract::TessdataManager::Init(const char *)': cannot convert argument 1 from 'const std:
:filesystem::path::value_type *' to 'const char *'`